lib/modules/manager/gradle/parser.spec.ts

```diff
@@ -724,6 +724,157 @@ describe('modules/manager/gradle/parser', () => {
         },
       ]);
     });
+
+    describe('content descriptors', () => {
+      it('valid combinations', () => {
+        const input = codeBlock`
+          maven(url = "https://foo.bar/baz") {
+            content {
+              excludeGroup("baz.qux")
+            }
+          }
+          mavenCentral().content {
+            includeGroup("foo.bar")
+          }
+          maven {
+            url = "https://foo.bar/deps"
+            content {
+              includeGroupAndSubgroups("foo.bar")
+            }
+          }
+          maven {
+            url = "https://some.foo"
+            content {
+              includeModule("foo", "bar")
+              excludeModule("baz", "qux")
+              includeVersion("foo", "bar", "1.2.3")
+              excludeVersion("baz", "qux", "4.5.6")
+              includeGroupByRegex("org\\\\.jetbrains\\\\.kotlin.*")
+              excludeGroupByRegex(".*google.*")
```

Comment:
is `google` the same as `/google/` or /^google$/`?

Reply:
gradle interprets `excludeGroupByRegex("google")` as `/^google$/`, but re2 unterstands it as `/google/`.

I think it makes sense to address this to avoid false positives. E.g., for `com.google.code.gson:gson:2.8.9` the pattern `com.google` matches in re2 but not in gradle, so renovate would wrongly exclude the dep. 

I'm not yet sure how to best catch this but probably with a pre-check, e.g., verify if the pattern matches `[\w.]+` and if yes, wrap it in `/^...$/`. Anyway thanks, very good catch!